### PR TITLE
[NFC][AMDGPU] clang-format `llvm/lib/Target/AMDGPU/SIISelLowering.cpp`

### PR DESCRIPTION
**Pull request by shiltian**

### CODE REVIEW

**Comment by shiltian** [comment]
* **#112645** <a href="https://app.graphite.dev/github/pr/llvm/llvm-project/112645?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @shiltian and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

**Comment by shiltian** [comment]
This is a pretty intrusive change. I can understand the reason if we don't want to do it, but the format of some code is really off.

File: llvm/lib/Target/AMDGPU/SIISelLowering.cpp [hunk 1]
```diff
@@ -3343,16 +3321,18 @@ void SITargetLowering::passSpecialInputs(
...
-    {AMDGPUFunctionArgInfo::QUEUE_PTR, "amdgpu-no-queue-ptr" },
-    {AMDGPUFunctionArgInfo::IMPLICIT_ARG_PTR, "amdgpu-no-implicitarg-ptr"},
-    {AMDGPUFunctionArgInfo::DISPATCH_ID, "amdgpu-no-dispatch-id"},
-    {AMDGPUFunctionArgInfo::WORKGROUP_ID_X, "amdgpu-no-workgroup-id-x"},
-    {AMDGPUFunctionArgInfo::WORKGROUP_ID_Y,"amdgpu-no-workgroup-id-y"},
-    {AMDGPUFunctionArgInfo::WORKGROUP_ID_Z,"amdgpu-no-workgroup-id-z"},
-    {AMDGPUFunctionArgInfo::LDS_KERNEL_ID,"amdgpu-no-lds-kernel-id"},
-  };
+                             StringLiteral>
+      ImplicitAttrs[] = {
```
Comment (arsenm) [inline llvm/lib/Target/AMDGPU/SIISelLowering.cpp:3325]: This is worse

File: llvm/lib/Target/AMDGPU/SIISelLowering.cpp [hunk 2]
```diff
@@ -4354,28 +4338,27 @@ SDValue SITargetLowering::lowerSET_FPENV(SDValue Op, SelectionDAG &DAG) const {
...
   Register Reg = StringSwitch<Register>(RegName)
-    .Case("m0", AMDGPU::M0)
-    .Case("exec", AMDGPU::EXEC)
-    .Case("exec_lo", AMDGPU::EXEC_LO)
-    .Case("exec_hi", AMDGPU::EXEC_HI)
-    .Case("flat_scratch", AMDGPU::FLAT_SCR)
-    .Case("flat_scratch_lo", AMDGPU::FLAT_SCR_LO)
-    .Case("flat_scratch_hi", AMDGPU::FLAT_SCR_HI)
-    .Default(Register());
+                     .Case("m0", AMDGPU::M0)
```
Comment (arsenm) [inline llvm/lib/Target/AMDGPU/SIISelLowering.cpp:4344]: Also worse

File: llvm/lib/Target/AMDGPU/SIISelLowering.cpp [hunk 3]
```diff
@@ -9849,11 +9809,11 @@ SDValue SITargetLowering::LowerINTRINSIC_VOID(SDValue Op,
     Ops.push_back(Op.getOperand(7 + OpOffset)); // imm offset
     unsigned Aux = Op.getConstantOperandVal(8 + OpOffset);
     Ops.push_back(
-      DAG.getTargetConstant(Aux & AMDGPU::CPol::ALL, DL, MVT::i8)); // cpol
+        DAG.getTargetConstant(Aux & AMDGPU::CPol::ALL, DL, MVT::i8)); // cpol
     Ops.push_back(DAG.getTargetConstant(
         Aux & AMDGPU::CPol::SWZ_pregfx12 ? 1 : 0, DL, MVT::i8)); // swz
-    Ops.push_back(M0Val.getValue(0)); // Chain
-    Ops.push_back(M0Val.getValue(1)); // Glue
+    Ops.push_back(M0Val.getValue(0));                            // Chain
```
Comment (arsenm) [inline llvm/lib/Target/AMDGPU/SIISelLowering.cpp:9815]: Don't know why it moved the comments all the way over there

File: llvm/lib/Target/AMDGPU/SIISelLowering.cpp [hunk 4]
```diff
@@ -10228,8 +10190,9 @@ SDValue SITargetLowering::handleByteShortBufferLoads(SelectionDAG &DAG,
     return DAG.getMergeValues({Value, Status, SDValue(Op.getNode(), 1)}, DL);
   }
 
-  unsigned Opc = (LoadVT.getScalarType() == MVT::i8) ?
-         AMDGPUISD::BUFFER_LOAD_UBYTE : AMDGPUISD::BUFFER_LOAD_USHORT;
+  unsigned Opc = (LoadVT.getScalarType() == MVT::i8)
```
Comment (arsenm) [inline llvm/lib/Target/AMDGPU/SIISelLowering.cpp:10193]: This can lose parentheses

File: llvm/lib/Target/AMDGPU/SIISelLowering.cpp (hunk 3 above)
Comment (shiltian) [reply llvm/lib/Target/AMDGPU/SIISelLowering.cpp:9815]: guess trying to align them

File: llvm/lib/Target/AMDGPU/SIISelLowering.cpp (hunk 2 above)
Comment (shiltian) [reply llvm/lib/Target/AMDGPU/SIISelLowering.cpp:4344]: This looks fine compared to the "previous worse" one, IMHO. At least it is aligned.

File: llvm/lib/Target/AMDGPU/SIISelLowering.cpp (hunk 1 above)
Comment (shiltian) [reply llvm/lib/Target/AMDGPU/SIISelLowering.cpp]: guarded by `clang-format off` now

File: llvm/lib/Target/AMDGPU/SIISelLowering.cpp [hunk 5]
```diff
@@ -5575,10 +5552,8 @@ MachineBasicBlock *SITargetLowering::EmitInstrWithCustomInserter(
     MachineBasicBlock *SplitBB = BB->splitAt(MI, false /*UpdateLiveIns*/);
     MachineBasicBlock *TrapBB = MF->CreateMachineBasicBlock();
     MF->push_back(TrapBB);
-    BuildMI(*TrapBB, TrapBB->end(), DL, TII->get(AMDGPU::S_ENDPGM))
-      .addImm(0);
-    BuildMI(*BB, &MI, DL, TII->get(AMDGPU::S_CBRANCH_EXECNZ))
-      .addMBB(TrapBB);
+    BuildMI(*TrapBB, TrapBB->end(), DL, TII->get(AMDGPU::S_ENDPGM)).addImm(0);
+    BuildMI(*BB, &MI, DL, TII->get(AMDGPU::S_CBRANCH_EXECNZ)).addMBB(TrapBB);
```
Comment (arsenm) [inline llvm/lib/Target/AMDGPU/SIISelLowering.cpp:5556]: clang-format always mangles BuildMI calls. I think all of these should be formatted with one add per line

Comment (shiltian) [reply llvm/lib/Target/AMDGPU/SIISelLowering.cpp:5556]: I don't think `clang-format` understands what the code "actually" means.

Comment (arsenm) [reply llvm/lib/Target/AMDGPU/SIISelLowering.cpp:5556]: Yes, it doesn't

Comment (shiltian) [reply llvm/lib/Target/AMDGPU/SIISelLowering.cpp:5556]: How do you suggest moving forward?

Comment (arsenm) [reply llvm/lib/Target/AMDGPU/SIISelLowering.cpp:5556]: Undo the cases where clang-format is wrong?

Comment (shiltian) [reply llvm/lib/Target/AMDGPU/SIISelLowering.cpp]: In order to make future life easier, I guarded all of them with `// clang-format off`.

File: llvm/lib/Target/AMDGPU/SIISelLowering.cpp [hunk 6]
```diff
@@ -16650,8 +16657,10 @@ bool SITargetLowering::requiresUniformRegister(MachineFunction &MF,
       for (auto &TC : TargetConstraints) {
         if (TC.Type == InlineAsm::isOutput) {
           ComputeConstraintToUse(TC, SDValue());
-          const TargetRegisterClass *RC = getRegForInlineAsmConstraint(
-              SIRI, TC.ConstraintCode, TC.ConstraintVT).second;
+          const TargetRegisterClass *RC =
+              getRegForInlineAsmConstraint(SIRI, TC.ConstraintCode,
+                                           TC.ConstraintVT)
+                  .second;
```
Comment (arsenm) [inline llvm/lib/Target/AMDGPU/SIISelLowering.cpp:16663]: I don't know why clang-format really wants to force things onto one line, and then sometimes produces these ugly breaks

**Review by arsenm — APPROVED** [review]

**Pull request merged**